retrofit/src/test/java/retrofit2/RequestFactoryTest.java

```diff
@@ -653,6 +654,46 @@
     }
   }
 
+  @Test public void getWithHeaders() {
+    class Example {
+      @GET("/search")
+      Call<ResponseBody> method(@HeaderMap okhttp3.Headers headers) {
```

Comment:
One last thing, can you also add a test that does both type to ensure they're combined instead of replaced?

Reply:
Good idea. Will also fix the style violation tomorrow.